### PR TITLE
docs: add performance debugging section v2

### PR DESCRIPTION
**Pull request by norg**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Add additional information about performance issue and how to debug those
- updated thanks to comments in https://github.com/OISF/suricata/pull/4525

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: doc/userguide/performance/analysis.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+In this section we will guide you through some options.
+
+General
+-------
+
+First of all you should check all the log files with a focus on stats.log and
+suricata.log if any obvious issues are seen. There are several tools that can
+help to find a root cause.
+
+A first step is to run a tool like **htop** to get an overview of the system
```
Comment (victorjulien) [inline doc/userguide/performance/analysis.rst]: Its a bit confusing that we have 'a first step' here, while the previous block said also that we need to do some other things 'First of all'.

File: doc/userguide/performance/analysis.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+If you see specific function calls at the top in red it's a hint that those are
+the bottlenecks. For example if you see **IPOnlyMatchPacket** it can be either
+a result of high drop rates or incomplete flows which result in decreased
+performance. To look into the performance issues on a specific thread you can
+pass **-t TID** to perf top.
+
+Another recommendation is to run Suricata without any rules to see if it's
+mainly related to the traffic. It can also be helpful to use rule-profiling
+and/or packet-profiling at this step. This is achieved by compiling Suricata
+with **enable-profiling** but keep in mind that this has an impact on
```
Comment (victorjulien) [inline doc/userguide/performance/analysis.rst]: `--enable-profiling`

File: doc/userguide/performance/analysis.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- Check for encapsulated traffic, while GRE, MPLS etc. are supported they could also lead to performance issues. Especially if there are several layers of encapsulation
+- Use tools like **iftop** to spot elephant flows. Flows that have a rate of over 1Gbit/s for a long time can result in one cpu core at 100% all the time and increasing the droprate while it doesn't make sense to dig deep into this traffic.
+- If VLAN is used it might help to disable **vlan.use-for-tracking** especially in scenarios where only one direction of the flow has the VLAN tag
+- If VLAN QinQ (IEEE 802.1ad) is used be very cautious if you use **cluster_qm** in combinatin with Intel drivers and AF_PACKET runmode. While the RFC expects ethertype 0x8100 and 0x88A8 in this case (see https://en.wikipedia.org/wiki/IEEE_802.1ad) most implementations only add 0x8100 on each layer. If the first seen layer has the same VLAN tag but the inner one has different VLAN tags it will still end up in the same queue in **cluster_qm** mode. This was observed with the i40e driver up to 2.8.20 and the firmare version up to 7.00, feel free to report if newer versions have fixed this.
```
Comment (victorjulien) [inline doc/userguide/performance/analysis.rst]: make 'report' a link to redmine or the suri support page?

File: doc/userguide/performance/analysis.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- Check for other unusual or complex protocols that aren't supported very well. In several cases we've seen that Cisco Fabric Path (ethertype 0x8903) causes performance issues. It's recommended to filter it, one option would be a bpf filter with **not ether proto 0x8903**
```
Comment (victorjulien) [inline doc/userguide/performance/analysis.rst]: this is new to me, where is this documented?

File: doc/userguide/performance/analysis.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- Another approach to narrow down issues is the usage of **bpf filter**. For example filter all HTTPS traffic with **not port 443** to exclude traffic that might be problematic or just look into one specific port **port 25** if you expect some issues with a specific protocol.
```
Comment (victorjulien) [inline doc/userguide/performance/analysis.rst]: add link to ignoring traffic page

**Comment by norg** [comment]
So you would like to have it restructured to look more like a workflow instead of a collection?
As the topics you listed are included, unless I've overseen one.

File: doc/userguide/performance/analysis.rst (hunk 5 above)
Comment (norg) [reply doc/userguide/performance/analysis.rst]: will do

File: doc/userguide/performance/analysis.rst (hunk 4 above)
Comment (norg) [reply doc/userguide/performance/analysis.rst]: I can create a redmine ticket for that in addition with more details.

File: doc/userguide/performance/analysis.rst (hunk 3 above)
Comment (norg) [reply doc/userguide/performance/analysis.rst]: later should be better so we have less places where we might have to update redmine links

File: doc/userguide/performance/analysis.rst (hunk 1 above)
Comment (norg) [reply doc/userguide/performance/analysis.rst]: good point, will rewrite it

File: doc/userguide/performance/analysis.rst (hunk 3 above)
Comment (victorjulien) [reply doc/userguide/performance/analysis.rst]: Not following. We can link to https://suricata-ids.org/support/ How would it be better to do it later?

**Comment by victorjulien** [comment]
> So you would like to have it restructured to look more like a workflow instead of a collection?
> As the topics you listed are included, unless I've overseen one.

Yeah. Right now it seems to jump back and forth between explain a tool (e.g. perf) and then suddenly treating something it may find (IPOnly..). I would like to see such things more clearly separated.

Btw with these tools like htop and perf (top) it would be helpful to add real world screenshots too.

**Comment by norg** [comment]
New PR will be created with the required changes included

**Pull request closed without merging**